xray/commands/curation/analyse.go

```diff
@@ -0,0 +1,170 @@
+package curation
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	utils2 "github.com/jfrog/jfrog-cli-core/v2/xray/commands/utils"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray/services/utils"
+	"net/http"
+	"strings"
+)
+
+type treeAnalyzer struct {
+	artiManager       artifactory.ArtifactoryServicesManager
+	artAuth           auth.ServiceDetails
+	httpClientDetails httputils.HttpClientDetails
+	url               string
+	repo              string
+	tech              coreutils.Technology
+}
+
+const (
+	BlockMessageKey  = "jfrog packages curation"
+	NotBeingFoundKey = "not being found"
+)
+
+func (nc *treeAnalyzer) recursiveNodeCuration(graph *utils.GraphNode, respStatus *[]PackageStatus, parent, parentVersion string, isRoot bool) error {
```

Comment:
I will add unit test, tnx

Reply:
I understand the need for recursion @asafambar.
How however are you dealing with duplicate depedencies in the graph? Do you need to send them multiple times, in case the same depedencies has multiple different parents?